scripts/packages.sh

```
@@ -0,0 +1,66 @@
+
+SCRIPTS_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "$SCRIPTS_DIR/detect-os.sh"
+
+# Helper function to filter package lists (removes comments and empty lines)
+filter_package_list() {
+    local file="$1"
+    grep -v '^#' "$file" | grep -v '^$'
+}
+
+install_packages() {
+    local PACKAGES_DIR="$(dirname "$SCRIPTS_DIR")/packages"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)